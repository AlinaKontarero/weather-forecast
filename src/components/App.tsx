import * as React from "react";
import CitySearch from "./CitySearch";
import ForecastLayout from "./ForecastsLayout";
import Footer from "./Footer";
import { Location, Coordinates } from "../types";
import { baseUrl } from "../utils/api-config";
import "../styles/App.css";

interface State {
  coords?: Coordinates
  locationWoeid?: number
}

class App extends React.Component<{}, State> {
  constructor(props: {}) {
    super(props)
    this.state = { 
      coords: undefined, 
      locationWoeid: undefined
    }
  }

  public componentDidMount() {
    if ("geolocation" in navigator) {
      navigator.geolocation.getCurrentPosition((position) =>  {
        this.setState({coords: position.coords}, async () => {
          if(!!this.state.coords) {
            const response = await fetch(`${baseUrl}/?lattlong=${this.state.coords.latitude},${this.state.coords.longitude}`)
            const possiblePlace = await response.json()
            if (possiblePlace.length > 0) {
              this.setState({ locationWoeid: possiblePlace[0].woeid });
              alert('Forecast is based on your browser geolocation data')
            }
          }
        })
      });
    } else {
      alert("Browser geolocation is not available");
    }
  }

  private  onSelectLocation = (value?: Location) => {
    this.setState({locationWoeid: value ? value.woeid : undefined });
  };

  public render() {
    return (
      <div className="App">
        <CitySearch onSelect={this.onSelectLocation} />
        {!!this.state.locationWoeid
          ? (<ForecastLayout 
              woeid={this.state.locationWoeid}
            />
          )
          : <div />
        }
        <Footer />
      </div>
  );
  }

};

export default App;
