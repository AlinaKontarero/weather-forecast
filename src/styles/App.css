:root {
  --white: #fff;
  --pink: #f93d66;
  --violet: #6d47d9;
  --text: #012b81;
  --background: #eaeaec;
  --primary: #f83c4b;
  --secondary: #38a8a9;
}

.App {
  text-align: center;
  background: url("../assets/Girl.png");
  background-position: right;
  background-repeat: no-repeat;
  height: 100vh;
  display: grid;
  grid-template-rows: 1fr 2fr 1fr;
  justify-content: center;
}
