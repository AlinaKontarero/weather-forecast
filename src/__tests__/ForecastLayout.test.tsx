import React from "react";
import { render } from "@testing-library/react";
import App from "../components/App";
import ForecastLayout from "../components/ForecastsLayout";

describe('ForecastLayout Component', () => {
  let component =  <ForecastLayout woeid={2122265} />
 
  beforeEach(() => {
    render(<App />);
    global.alert = jest.fn();
    });  
  
  it('should render with given props', () => {
    expect(component).toMatchSnapshot();
  });
});