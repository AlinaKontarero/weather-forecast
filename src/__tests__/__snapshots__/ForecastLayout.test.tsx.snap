// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastLayout Component should render with given props 1`] = `
<ForecastLayout
  woeid={2122265}
/>
`;
